### PR TITLE
enable linux build and test using Travis

### PR DESCRIPTION
**Pull request by shiftkey**
After uncovering that I'd broken the app completely for Linux users in #2807, I'd like to get us back and running tests on Travis against the Linux version so we catch these earlier.

I've disabled publishing for now to ensure we don't accidentally publish things, but once we start exploring the packaging stuff again in #2300 and ensure our update server can consume new releases we can start talking about what official support looks like.

### CODE REVIEW

**Comment by shiftkey** [comment]
@joshaber I couldn't spot the right option to re-enable builds on Travis. You mentioned disabling this when we switched over to CircleCI. Mind doing the needful if we're 👍  on this PR?

**Comment by shiftkey** [comment]
An example of this successful build: https://travis-ci.org/shiftkey/desktop/builds/278022535

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,24 @@ notifications:
...
+matrix:
+  include:
+    - os: linux
+      env: CC=clang CXX=clang++ npm_config_clang=1
+      compiler: clang
+
+addons:
+  apt:
+    packages:
+      # by default the container VMs have Git 1.9.1, ermagherd...
```
Comment (joshaber) [inline .travis.yml]: 😱

**Comment by joshaber** [comment]
💃

**Pull request merged**